Chrysalis-Papilio/Soen490-Papilio-Website
# PAPILIO-38-story

## General

Without statistics of their activities and their ad campaigns, companies would not be able to make the appropriate decision on our application. To help them, we will develop a mechanism that will capture information regarding user activity on the app to really understand what they are doing in those companies activities and how they are interacting with them. 

With this information, companies will be able to make better decision on how they are creating their activity pages and the groups they are targeting in their ads. In the website, this will be shown in graph that will be present in the main page of the Dashboard so that companies won't have to search for them across the app. It is some of the most important information in the dashboard which explain why we positioned it there.

_Issue number_: Papilio-38

## Task description
- Create custom parameter in Firebase analytics
- Create analytics database
- Create cron job to get information from analytics to database 
- Connect analytics backend to website dashboard.
- Add associated tests

## Manual Test

Make sure that you have the appropriate variables in the backend .env file. (Contact me if you don't)
Since we don't have much data accumulated, there is a possibility that the company does not have enough information to fill the graph. Contact me if this is the case.

1. start the backend
`npm run build && npm start`
2. start the website (it should be running to prove the npm conflict solution)
`npm start`
3. login to the dashboard
4. You should see the graph with some statistics
5. You can specify activity with the dropdown on the top right
6. Graph should now be specific to that activity

## Testing
To run all the unit tests, run the following code: npm test
To make sure that the tests have reach the appropriate level of coverage, run the following code: npm run coverage
